### ISSUE TITLE
:<- macro does not work across namespaces

### ISSUE THREAD

**Issue by joshkh** [Day8/re-frame#207]
In the following example, using the `:<-` macro in the `myapp.components.events` namespace causes re-frame to throw an exception stating that the `:main/some-items` subscription handler cannot be found

```cljs
(ns myapp.core
  (:require [myapp.events]
            [myapp.component.events]))
```

```cljs
(ns myapp.events
  (:require [re-frame.core :refer [reg-sub]]))

(reg-sub
  :main/some-items
  (fn [db]
    (:all-items db)))
```

```cljs
(ns myapp.component.events
  (:require [re-frame.core :refer [reg-sub]]))

(reg-sub
  :components/filtered-items
  :<- [:main/some-items]
  (fn [items]
    (filter odd? items)))
````

> re-frame: no subscription handler registered for: " cljs.core.Keyword_hash: 1803312489cljs$lang$protocol_mask$partition0$: 2153775105cljs$lang$protocol_mask$partition1$: 4096fqn: "main/some-items"name: "some-items"ns: "main"__proto__: Object ". Returning a nil subscription.cljs.core.apply.cljs$core$IFn$_invoke$arity$2 @ core.cljs:3679
subs.cljc?rel=1472037082624:64Uncaught TypeError: Cannot read property 'call' of null

**Comment by stumitchell** [comment]
I would make sure that `myapp.events` is loaded before `myapp.component.events` to do this change the require in `myapp.component.events` to the following.

    (ns myapp.component.events
      (:require [re-frame.core :refer [reg-sub]]
                [myapp.events]))

Let us know if that fixes it.

**Comment by mike-thompson-day8** [comment]
@joshkh  Given the error message you posted, you really, really need to be using clj-devtools https://github.com/binaryage/cljs-devtools.

**Comment by mike-thompson-day8** [comment]
I'm about 100% sure Stu has answered your question.  So I'm going to close this.  Will reopen as necessary, but please direct support questions to clojurians.

**Comment by jamesnvc** [comment]
Something I'd like to add to this issue is that reg-sub is inconsistent as to when this will and will not work:  Specifically, as in the example above, with one pair it fails, but if you have multiple pairs of `:<- [:sub]` it does work (since the implementation uses `map` in the multiple-sub cases, so the value isn't realized until later).  It is very confusing to have something work and then stop working when removing subs - I submit it would be better if it either worked (preferable) or didn't consistently.

**Comment by danielcompton** [comment]
@jamesnvc that behaviour is fixed in #259, the behaviour of the single pair now matches the multiple pairs.

**Comment by jamesnvc** [comment]
@danielcompton perfect, thank you so much for the prompt reply!
